evalml/tests/data_checks_tests/test_target_distribution_data_check.py

```diff
@@ -0,0 +1,170 @@
+                message_code=DataCheckMessageCode.TARGET_IS_NONE,
+                details={},
+            ).to_dict()
+        ],
+        "actions": [],
+    }
+
+
+@pytest.mark.parametrize("target_type", ["boolean", "categorical", "integer", "double"])
+def test_target_distribution_data_check_unsupported_target_type(
+    target_type, X_y_regression
+):
+    X, y = X_y_regression
+
+    if target_type == "boolean":
+        y = pd.Series([True, False] * 50)
+    elif target_type == "categorical":
+        y = pd.Series(["One", "Two", "Three", "Four"] * 25)
+    elif target_type == "integer":
+        y = pd.Series(np.random.randint(1, 25, 100))
+    else:
+        y = np.random.normal(4, 1, 100)
+
+    y = infer_feature_types(y)
+
+    target_dist_check = TargetDistributionDataCheck("regression")
+
+    if target_type in ["integer", "double"]:
+        assert target_dist_check.validate(X, y) == {
+            "warnings": [],
+            "errors": [],
+            "actions": [],
+        }
+    else:
+        assert target_dist_check.validate(X, y) == {
+            "warnings": [],
+            "errors": [
+                DataCheckError(
+                    message=f"Target is unsupported {y.ww.logical_type.type_string} type. Valid Woodwork logical types include: integer, double",
+                    data_check_name=target_dist_check_name,
+                    message_code=DataCheckMessageCode.TARGET_UNSUPPORTED_TYPE,
+                    details={"unsupported_type": y.ww.logical_type.type_string},
+                ).to_dict()
+            ],
+            "actions": [],
+        }
+
+
+@pytest.mark.parametrize("problem_type", ["binary", "multiclass", "regression"])
+def test_target_distribution_data_check_invalid_problem_type(
+    problem_type, X_y_regression
+):
+    X, y = X_y_regression
+
+    target_dist_check = TargetDistributionDataCheck(problem_type)
+
+    if problem_type == "regression":
+        assert target_dist_check.validate(X, y) == {
+            "warnings": [],
+            "errors": [],
+            "actions": [],
+        }
+    else:
+        assert target_dist_check.validate(X, y) == {
+            "warnings": [],
+            "errors": [
+                DataCheckError(
+                    message=f"Problem type {problem_type} is unsupported. Valid problem types include: [ProblemTypes.REGRESSION, ProblemTypes.TIME_SERIES_REGRESSION]",
+                    data_check_name=target_dist_check_name,
+                    message_code=DataCheckMessageCode.TARGET_UNSUPPORTED_PROBLEM_TYPE,
+                    details={
+                        "unsupported_problem_type": handle_problem_types(problem_type)
+                    },
+                ).to_dict()
+            ],
+            "actions": [],
+        }
+
+
+@pytest.mark.parametrize("data_type", ["positive", "mixed", "negative"])
+@pytest.mark.parametrize("distribution", ["normal", "lognormal", "lognormal_"])
+def test_target_distribution_data_check_warning_action(
```

Comment:
So with `lognormal` vs `lognormal_`, is the difference that one is much more lognormal (`lognormal_`) compared to the other? I don't understand why having the `p_value=0` is important here, since I assume the `p_value` is low enough to reject the null hypothesis with `lognormal`?

Reply:
I've updated it with a comment that clarifies it. Basically this is meant to check the `=` part of `shapiro_test_log.pvalue >= shapiro_test_og.pvalue`. In the event that the target data is log transformed and doesn't reduce its `p-value`, it falls under the assumption that it's as normal if not more `normal` than before. That more or less implies that the underlying distribution had some log normality to it. The `p-value` is low enough that we can reject the null hypothesis that the distribution is `normal`, but a distribution that maintains or increases its `p-value` post log transformation would give us enough of a reason to believe that log normality exists.

Reply:
Thanks for the clarification!